File: cPlusPlus/source/block.cpp
#include "../headers/block.h"

Block::Block(Document& d, string pH, int i)
    : data(d), previousHash(pH), index(i){
    this->timestamp = system_clock::to_time_t(system_clock::now());
    this->nonce = 0;
    this->setHash();
}

string Block::getHash(){
    return this->hash;
}

string Block::getPreviousHash(){
    return this->previousHash;
}

string Block::getTimestampAsString(){
    stringstream ss;
    ss << put_time(localtime(&this->timestamp), "%d/%m/%y - %T");
    return ss.str();
}

string Block::getAttributesAsString(){
    return to_string(this->index) + StringUtil::documentToString(this->data) + this->previousHash +
        this->getTimestampAsString() + to_string(this->nonce);
}

void Block::setHash(){
    this->hash = StringUtil::applySha256(this->getAttributesAsString());
}

void Block::showFullData(){
    cout << "Índice        : " << this->index << endl;
    cout << "Dados         : " << StringUtil::documentToString(this->data) << endl;
    cout << "Hash Anterior : " << this->previousHash << endl;
    cout << "Timestamp     : " << this->getTimestampAsString() << endl;
    cout << "Nonce         : " << this->nonce << endl;
    cout << "Hash Atual    : " << this->hash << "\n" << endl;
}

File: oldPythonFiles/block.py
from datetime import datetime
from hashlib import sha256

class Block:

    def __init__(self, data, previousHash, index):

        self.nonce = 0 #int
        self.data = data #dict
        self.previousHash = previousHash #string
        self.index = index #int
        self.timestamp = datetime.now() #date
        self.setHash() #hex string

    def getAttributesAsString(self):
        return (self.data + str(self.index) + self.timestamp.strftime("%d/%m/%y - %T") +
                self.previousHash + str(self.nonce))

    def getAttributesTypes(self):
        return (str(type(self.nonce)) +
                str(type(self.data))+
                str(type(self.previousHash)) +
                str(type(self.index)) +
                str(type(self.timestamp)) +
                str(type(self.myHash) ))

    def showFullData(self):
        print("Índice        : " + str(self.index))
        print("Timestamp     : " + self.timestamp.strftime("%d/%m/%y - %T"))
        print("Dados         : " + self.data)
        print("Nonce         : " + str(self.nonce))
        print("Hash Anterior : " + self.previousHash)
        print("Hash Atual    : " + self.myHash+"\n")

    def setHash(self):
        self.myHash = sha256(self.getAttributesAsString().encode()).hexdigest()